validator/client/runner.go

```diff
@@ -39,13 +36,13 @@ func run(ctx context.Context, v iface.Validator) {
 	cleanup := v.Done
 	defer cleanup()
 
-	headSlot, err := initializeValidatorAndGetHeadSlot(ctx, v)
-	if err != nil {
-		return // Exit if context is canceled.
-	}
-	if err := v.UpdateDuties(ctx, headSlot); err != nil {
-		handleAssignmentError(err, headSlot)
+	if err := v.Init(ctx); err != nil {
+		if errors.Is(err, context.Canceled) {
+			return // Exit if context is canceled.
```

Comment:
Are you sure we should continue for all other errors? For example an error in `WaitForKeymanagerInitialization` is critical, no?

Reply:
this should just safely leave the entire run not skip

Reply:
in the case of the log below it's fatal so it shouldn't be ignoring